packages/salesforcedx-vscode-soql/src/editor/soqlEditorProvider.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as fs from 'fs';
+import * as path from 'path';
+import * as vscode from 'vscode';
+
+export class SOQLEditorProvider implements vscode.CustomTextEditorProvider {
+  public static register(context: vscode.ExtensionContext) {
+    const provider = new SOQLEditorProvider(context);
+    const providerRegistration = vscode.window.registerCustomEditorProvider(
+      SOQLEditorProvider.viewType,
+      provider
+    );
+    return providerRegistration;
+  }
+
+  private static readonly viewType = 'soqlCustom.soql';
+  constructor(private readonly context: vscode.ExtensionContext) {}
+
+  public async resolveCustomTextEditor(
+    document: vscode.TextDocument,
+    webviewPanel: vscode.WebviewPanel,
+    _token: vscode.CancellationToken
+  ): Promise<void> {
+    webviewPanel.webview.options = {
+      enableScripts: true,
+      localResourceRoots: [
+        vscode.Uri.file(
+          path.join(
+            this.context.extensionPath,
+            'node_modules/@salesforce/soql-builder-ui'
```

Comment:
will this path stay the same on non-dev environments too?

Reply:
We should pull this string out as a constant.

Reply:
yes I think this will stay the same regardless because the UI is listed as a dependency of the extension.

I will pull this out as a contant

Reply:
@praksb See the `use constants for file paths` commit.